### PR TITLE
Fix whitespace diff option button

### LINKED ISSUE #322: Fix button for diffs without whitespace
Since the new code reviews functionality on GitHub there has been some interface changes. GitHub changed the diff options from a drop down to a button group so the feature initially implemented in #196 stopped working.

We should change the code putting the button to make checks and modifications against the updated DOM structure.

---

The key to fixing this is changing how we check for the current DOM here: https://github.com/sindresorhus/refined-github/blob/b12ad0b07d24602fe400d887004540734b742500/extension/content.js#L240 and how we insert the button here: https://github.com/sindresorhus/refined-github/blob/b12ad0b07d24602fe400d887004540734b742500/extension/content.js#L269-L275

### PR DESCRIPTION
**Pull request by anselmh**
This fixes #322.

TL;DR: Github changed the markup from a select optionbox
to a tablist. This change re-implements the whitespace
diff toggle option as third tab-item on diff view pages.

## Screenshots of new layout:
screen shot 2017-05-13 at 17 20 39
screen shot 2017-05-13 at 17 20 47

### CODE REVIEW

**Comment by hkdobrev** [comment]
@anselmh Awesome! Thanks! Could you also please add the button to commit pages? https://github.com/sindresorhus/refined-github/commit/e90c193defaf020b32d3a6f5b3b21ad4bcb6a0e6

**Comment by anselmh** [comment]
@hkdobrev done now.

**Comment by hkdobrev** [comment]
image

In some cases, like the one above there's not enough space to fit everything with the new button.

Here's a link: https://github.com/sindresorhus/refined-github/pull/395/files/4a7c64cd0d3d3103a5307b11fb1232490d119867..c0c2913435837d0d0cc08800ef56af7f3a634f69

Could we make the button text shorter? E.g. "No whitespace". Or use an icon perhaps?

**Comment by anselmh** [comment]
I’ve updated the PR again. Not sure this is better but it now has a shorter title:
screen shot 2017-05-14 at 11 11 34
screen shot 2017-05-14 at 11 11 46

**Comment by jgierer12** [comment]
I think we should just make this a separate button altogether. It's not really related to "Unified"/"Split". And maybe make it a checkbox instead: 
Show whitespace concept

File: extension/content.js [hunk 1]
```diff
@@ -281,22 +281,23 @@ function addDiffViewWithoutWhitespaceOption(type) {
...
 	} else {
 		urlParams.set('w', 1);
 	}
 
-	const optionElementContent = `${optionIsSet ? svgIcon : ''} Ignore whitespace`;
+	const optionElementContent = `${optionIsSet ? svgIcon + ' ' : ''}No Whitespace`;
 	const optionHref = `${window.location.origin + window.location.pathname}?${urlParams.toString() + urlHash}`;
 
-	optionElementObject.html(optionElementContent).attr('href', optionHref).attr('data-hotkey', 'd w').attr('class', 'refined-github-toggle-whitespace');
+	optionElementObject.html(optionElementContent).attr('href', optionHref).attr('data-hotkey', 'd w').attr('class', 'refined-github-toggle-whitespace').attr('title', `Click here to turn whitespace in diffs ${optionIsSet ? 'on' : 'off'}`);
```
Comment (jgierer12) [inline extension/content.js]: I'd prefer to build the element using a template string (e.g. https://github.com/sindresorhus/refined-github/blob/master/extension/content.js#L328-L332)

Comment (jgierer12) [inline extension/content.js]: I'd prefer building the element using a template string (e.g. https://github.com/sindresorhus/refined-github/blob/master/extension/content.js#L328-L332)

File: extension/content.js [hunk 2]
```diff
@@ -503,7 +504,7 @@ document.addEventListener('DOMContentLoaded', () => {
 
 			if (pageDetect.isPRFiles() || pageDetect.isPRCommit()) {
 				diffFileHeader.setup();
-				addDiffViewWithoutWhitespaceOption('pr');
+				addDiffViewWithoutWhitespaceOption();
```
Comment (jgierer12) [inline extension/content.js]: Duplicate function call (https://github.com/sindresorhus/refined-github/pull/425/files#diff-a8a1389cea3a1a117e5372bd080a5926R498)

File: extension/content.js [hunk 3]
```diff
@@ -261,8 +261,8 @@ function showRecentlyPushedBranches() {
 }
 
 // Add option for viewing diffs without whitespace changes
-function addDiffViewWithoutWhitespaceOption(type) {
-	if ($('.diff-options-content').length < 1 && $('.btn-group .selected[href*="diff="]').length < 1) {
+function addDiffViewWithoutWhitespaceOption() {
+	if ($('.pr-review-tools .BtnGroup').length < 1 && $('.Details .BtnGroup').length < 1 && $('.pr-review-tools .BtnGroup .tooltipped[href*="diff="]').length < 1) {
```
Comment (jgierer12) [inline extension/content.js]: Store `$('.pr-review-tools .BtnGroup')` and `$('.Details .BtnGroup')` into variables, they're used 3 times each.

**Comment by jgierer12** [comment]
@anselmh ping 🔔

File: extension/content.js (hunk 1 above)
Comment (anselmh) [inline extension/content.js]: Hey, to be honest I have no idea how to do this as the attributes are set on `optionElementObject`, not on a child that’s injected via `html()`? If you could help me out here, I can try doing that.

**Comment by anselmh** [comment]
Done all requested changes and altered the layout to have this a separate button.

screen shot 2017-05-31 at 11 29 50
screen shot 2017-05-31 at 11 44 46

**Comment by jgierer12** [comment]
👍 , but I'd still prefer this to be a checkbox. @hkdobrev what do you think?

**Comment by sindresorhus** [comment]
I think it's good. It has a checkbox when the button is clicked. It fits in better now than with a plain checkbox.

**Comment by sindresorhus** [comment]
Thank you @anselmh :)
